### DISCUSSION TITLE
请问如何识别客户端

### DISCUSSION THREAD

**Discussion by jersey44168** [AdguardTeam/AdGuardHome#5905] [Q&A] [answered]
先决条件
1、我使用的是adguard桌面版本来添加加密dns
2、有添加通配符证书文件
3、有开放服务器端口

我尝试了教程：https://github.com/AdguardTeam/AdGuardHome/wiki/Clients#clientid

其中：DNS-over-HTTPS： .https://example.org/dns-query/my-client  这个方式可以添加识别客户端、

剩下的三个都不能添加：

https://my-client.example.org/dns-query

quic://my-client.example.org

tls://my-client.example.org

**Comment by ainar-g** [comment, marked as answer]
You need to make sure that:

1.  Your certificate is a wildcard certificate.  That is, that it actually covers domain names like `my-client.example.org`.

2.  The domain names are actually resolvable by your client.  That is, you can query `my-client.example.org` on the client machine and receive an accessible IP address of your AdGuard Home.

3.  The certificate must be valid for the client (or, clients should be able to skip verification, if you're using AGH for a local setup).

Inspect the verbose logs of both server and client (if there are ways to do it there) to see if there are any errors related to certificates or domains that cannot be resolved.

**Reply by jersey44168** [reply]
thanks buddy,i found my certificate isn't a wildcard certificate, and i deal with it.
